src/main/java/com/parksupark/soomjae/server/fcm/service/AlarmNotificationService.java

```diff
@@ -0,0 +1,183 @@
+package com.parksupark.soomjae.server.fcm.service;
+
+import com.parksupark.soomjae.server.common.exception.ErrorMessages;
+import com.parksupark.soomjae.server.community.comment.entity.Comment;
+import com.parksupark.soomjae.server.community.comment.repository.CommentRepository;
+import com.parksupark.soomjae.server.community.common.constant.PostConstant;
+import com.parksupark.soomjae.server.community.common.exception.InvalidPostIdException;
+import com.parksupark.soomjae.server.community.like.entity.Like;
+import com.parksupark.soomjae.server.community.participation.entity.Participation;
+import com.parksupark.soomjae.server.community.post.communitypost.repository.CommunityPostRepository;
+import com.parksupark.soomjae.server.community.post.meetingpost.repository.MeetingPostRepository;
+import com.parksupark.soomjae.server.community.validator.PostValidator;
+import com.parksupark.soomjae.server.community.validator.PostValidatorFactory;
+import com.parksupark.soomjae.server.fcm.domain.Token;
+import com.parksupark.soomjae.server.fcm.dto.AlarmDto;
+import com.parksupark.soomjae.server.fcm.repository.TokenRepository;
+import com.parksupark.soomjae.server.member.entity.Member;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class AlarmNotificationService {
+
+    private final PostValidatorFactory postValidatorFactory;
+    private final CommentRepository commentRepository;
+    private final TokenRepository tokenRepository;
+    private final CommunityPostRepository communityPostRepository;
+    private final MeetingPostRepository meetingPostRepository;
+    private final FCMService fcmService;
+
+    @Transactional
+    public void sendNewCommentAlarm(Comment comment) {
+
+        Member writer = comment.getMember();
+
+        List<Member> receivers = new ArrayList<>();
+
+        // 댓글 단 사람들 추가
+        List<Member> commenters = commentRepository.findMembersByPost(comment.getPostType(),
+            comment.getPostId());
+
+        receivers.addAll(commenters);
+
+        // 2) 중복 제거 + 이번에 댓글 쓴 사람 제외
+        receivers = receivers.stream()
+            .distinct()
+            .filter(m -> !m.getId().equals(writer.getId()))
+            .toList();
```

Comment:
```suggestion
        receivers = commenters.stream()
            .distinct()
            .filter(m -> !m.getId().equals(writer.getId()))
            .toList();
```

commenters로 바로 filter하면 안되나요? 🤔

Reply:
본인이 작성한 댓글은 알림을 받지 않도록 하기 위해서입니다!

Reply:
@haha4825 receivers.addAll한 이후 다시 필터링 하는 것이 문제라고 생각되는 점입니다.